src/OpenTelemetry.AutoInstrumentation/Configuration/Settings.cs

```diff
@@ -0,0 +1,85 @@
+// <copyright file="Settings.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using OpenTelemetry.Exporter;
+
+namespace OpenTelemetry.AutoInstrumentation.Configuration
+{
+    /// <summary>
+    /// Global Settings
+    /// </summary>
+    public abstract class Settings
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Settings"/> class
+        /// using the specified <see cref="IConfigurationSource"/> to initialize values.
+        /// </summary>
+        /// <param name="source">The <see cref="IConfigurationSource"/> to use when retrieving configuration values.</param>
+        protected Settings(IConfigurationSource source)
+        {
+            if (source == null)
+            {
+                throw new ArgumentNullException(nameof(source));
+            }
+
+            OtlpExportProtocol = GetExporterOtlpProtocol(source);
+            Http2UnencryptedSupportEnabled = source.GetBool(ConfigurationKeys.Http2UnencryptedSupportEnabled) ?? false;
+            FlushOnUnhandledException = source.GetBool(ConfigurationKeys.FlushOnUnhandledException) ?? false;
+            Integrations = new IntegrationSettingsCollection(source);
+        }
+
+        /// <summary>
+        /// Gets the the OTLP transport protocol. Supported values: Grpc and HttpProtobuf.
+        /// </summary>
+        public OtlpExportProtocol? OtlpExportProtocol { get; }
+
+        /// <summary>
+        /// Gets a collection of <see cref="Integrations"/> keyed by integration name.
+        /// </summary>
+        public IntegrationSettingsCollection Integrations { get; }
```

Comment:
The backing collection of integrations is sourced from the `IntegrationRegistry`, which is populated by the values from the `TracerInstrumentation` enum. Did you want to combine the metrics integrations with the tracer integrations, or keep them separate?

Reply:
At this point, no integrations are needed for metrics. I want to keep them separate when use case arises for it.

Reply:
Moved `IntegrationSettingsCollection` to `TracerSettings`.